### DISCUSSION TITLE
OpenTelemetry and Kafka - Create span of Consumer on annotation of Incoming

### DISCUSSION THREAD

**Discussion by arik-dig** [quarkusio/quarkus#30688] [Q&A]
The standard of OTEL for messaging suggests that Spans of [Consumer kind](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md#consumer) should be opened whenever message is consumed.
I see in the [quick-guide of kafka](https://quarkus.io/guides/kafka-reactive-getting-started) ([git link](https://github.com/quarkusio/quarkus-quickstarts/tree/main/kafka-quickstart)) that best practice for consuming messages is to use the annotation of `org.eclipse.microprofile.reactive.messaging.Incoming`

are you planning to add this kind of instrumentation - [Consumer Kind](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md#consumer) on method annotated with \@incoming?

[a comment by an automated account is not shown]

**Comment by arik-dig** [comment]
I saw similar discussions like #26353 but nothing related to opening a new span of Consumer Kind

**Reply by brunobat** [reply]
I Believe this PR has added support for that, no? https://github.com/smallrye/smallrye-reactive-messaging/pull/1678/

**Reply by radcortez** [reply]
Yes, the PR should have proper instrumentation support for messaging, but it is not available in Quarkus yet, I believe.

Can you confirm when are we updating the SmallRye RM version @ozangunalp ?

**Reply by ozangunalp** [reply]
Yes it is ready to go. But I have been unable to publish to the maven central since a couple of days. Sonatype is having issues.

**Reply by arik-dig** [reply]
How soon is it going to be part of Quarkus?
I would be happy to be notified somehow when it does happen.
so will you update it in here in this discussion?
or should I subscribe to some other discussion and/or PR?

Thanks

**Comment by arik-dig** [comment]
@radcortez , @brunobat are you planning to put attributes of `code.funtion` and `code.namespace` based on the annotation `org.eclipse.microprofile.reactive.messaging.Incoming` ?
it is a common recommended [semantic convention](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/span-general.md#source-code-attributes)

**Reply by brunobat** [reply]
Created this: https://github.com/quarkusio/quarkus/issues/30793
